app/routines/search_users.rb

```diff
@@ -36,33 +40,51 @@ class SearchUsers
   SORT_DESCENDING = 'DESC'
   MAX_MATCHING_USERS = 10
 
+  def matches(field)
+    debugger
+    field.like_any
+  end
+
   def exec(query, options={})
 
     users = User.scoped
-    
+
     KeywordSearch.search(query) do |with|
 
-      with.default_keyword :any
+      match_method = options[:exact] ? :any : :like_any
 
-      with.keyword :username do |usernames|
-        users = users.where{username.like_any my{prep_names(usernames)}}
-      end
+      compose_clause = if options[:exact]
+                         lambda{ |q,field,names| q.lower(field).any prep_names(names) }
+                       else
+                         lambda{ |q,field,names| q.lower(field).like_any prep_names(names) }
+                       end
 
       with.keyword :first_name do |first_names|
-        users = users.where{lower(first_name).like_any my{prep_names(first_names)}}
+         users = users.where{
+           lower(first_name).send(match_method, my{prep_names(first_names)})
+         }
+      end
+
+      with.default_keyword :any
+      with.keyword :username do |usernames|
+        users = users.where{ | q | compose_clause[q, q.username, usernames] }
       end
```

Comment:
This is my preferred way of doing it, but thought it might not be immediately obvious what I was doing. It sets up a lamba that's assigned to the `compose_clause` variable and then calls it for the query.